samples/sample_decode/src/pipeline_decode.cpp

```diff
@@ -1660,21 +1655,18 @@ mfxStatus CDecodingPipeline::RunDecoding()
     bool                bErrIncompatibleVideoParams = false;
     CTimeInterval<>     decodeTimer(m_bIsCompleteFrame);
     time_t start_time = time(0);
-    MSDKThread * pDeliverThread = NULL;
 #if (MFX_VERSION >= 1025)
     mfxExtDecodeErrorReport *pDecodeErrorReport = NULL;
 #endif
 
-    if (m_eWorkMode == MODE_RENDERING) {
+    if (m_eWorkMode == MODE_RENDERING)
+    {
         m_pDeliverOutputSemaphore = new MSDKSemaphore(sts);
         m_pDeliveredEvent = new MSDKEvent(sts, false, false);
-        pDeliverThread = new MSDKThread(sts, DeliverThreadFunc, this);
-        if (!pDeliverThread || !m_pDeliverOutputSemaphore || !m_pDeliveredEvent) {
-            MSDK_SAFE_DELETE(pDeliverThread);
-            MSDK_SAFE_DELETE(m_pDeliverOutputSemaphore);
-            MSDK_SAFE_DELETE(m_pDeliveredEvent);
-            return MFX_ERR_MEMORY_ALLOC;
-        }
+
+        // Create a thread and let it work independently from the thread handle 
+        std::thread deliverThread(&CDecodingPipeline::DeliverLoop, this);
+        deliverThread.detach();
```

Comment:
hm. why we detach the thread? what is your motivation for that? Will this make memory leaks checking more complicated? IMHO, I would prefer to see honest join in the very end.

Reply:
Well, ```join``` looks more similar to original code, so I redesigned it with ```join```